### PR TITLE
Implemented: support to reorder pending export jobs from the updates page(#2dmggua)

### LINKED ISSUE #72: Add UI to drag and reorder threshold rules
Users should be able to drag and drop threshold rules on the "Threshold updates" page to reorder them. Functionality to update threshold sequences in the backend will be done separately.

### PR DESCRIPTION
**Pull request by ymaheshwari1**
### Related Issues
 

 Closes #72 

 ### Short Description and Why It's Useful
 

 ### Screenshots of Visual Changes before/after (If There Are Any)
 

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/components/JobReorderModal.vue [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    IonItem,
+    IonLabel,
+    IonReorder,
+    IonReorderGroup,
+    IonTitle,
+    IonToolbar,
+  },
+  data() {
+    return {
+      updatedJobsOrder: [] as any,
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue:78]: We are using it inside save() method, I think we should not make it data property

File: src/components/JobReorderModal.vue [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      jobs: (this as any).jobsForReorder,
+      seqBeforeReorder: JSON.parse(JSON.stringify((this as any).jobsForReorder))
+    }
+  },
+  computed: {
+    ...mapGetters({
+      userProfile: "user/getUserProfile"
+    }),
+  },
+  props: ["jobsForReorder"],
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue:90]:
```suggestion
  props: ["jobs"],
```

File: src/components/JobReorderModal.vue [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    IonReorderGroup,
+    IonTitle,
+    IonToolbar,
+  },
+  data() {
+    return {
+      updatedJobsOrder: [] as any,
+      failedJobs: [] as any,
+      successJobs: [] as any,
+      jobs: (this as any).jobsForReorder,
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue:81]: We could have modifiedJobs as clone copy of jobs & updated jobs and seqBeforeReorder might not be needed as the jobs property has this value

File: src/components/JobReorderModal.vue [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+          const resp = await JobService.updateJob(payload)
+          if (resp.status == 200 && !hasError(resp) && resp.data.successMessage) {
+            // if the job succeded when updating then adding the jobId to the successJobs array
+            this.successJobs.push(job.jobId)
+            showToast(translate('Jobs sequence updated successfully'))
+          } else {
+            // if the job failed when updating then adding the jobId to the failedJobs array
+            this.failedJobs.push(job.jobId)
+            logger.error('Failed to update some jobs')
+            showToast(translate('Failed to update some jobs'))
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue:154]: I think this should be done once after all the iterations, if all success we show success message else the error one

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobReorderModal.vue (hunk 1 above)
Comment (ymaheshwari1) [reply src/components/JobReorderModal.vue:78]: Sir, we will be using this variable in updating the runTime of jobs, that's why made it local data property.

File: src/components/JobReorderModal.vue (hunk 4 above)
Comment (ymaheshwari1) [reply src/components/JobReorderModal.vue:154]: Yes Sir

File: src/components/JobReorderModal.vue (hunk 2 above)
Comment (ymaheshwari1) [reply src/components/JobReorderModal.vue:90]: Ok Sir

File: src/components/JobReorderModal.vue [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+          [key]: updatedSeq[key]
+        }
+      }, {})
+      return diffSeq;
+    },
+    doReorder(event: CustomEvent) {
+      // making the item reorder action as complete and storing the updated order in jobs
+      const updatedSeq = event.detail.complete(JSON.parse(JSON.stringify(this.modifiedJobs)));
+      let diffSeq = this.findJobDiff(this.jobs, updatedSeq)
+      const updatedRunTime = this.jobs.map((job: any) => job.runTime)
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue]: Improve variable name

File: src/components/JobReorderModal.vue [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+      Object.keys(diffSeq).map((key: any) => {
+        diffSeq[key].runTime = updatedRunTime[key]
+      })
+      diffSeq = Object.keys(diffSeq).map((key) => diffSeq[key])
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue]:
```suggestion
      diffSeq = Object.keys(diffSeq).map((key: any) => {
        diffSeq[key].runTime = updatedRunTime[key]
        return diffSeq[key]
      })
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobReorderModal.vue [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+        <ion-label>{{ job.runTime ? getTime(job.runTime) : "-"  }}</ion-label>
+        <ion-label>{{ timeTillJob(job.runTime)}}</ion-label>
+        <ion-icon :icon="checkmarkCircleOutline" color="success" v-if="successJobs.includes(job.jobId)" />
+        <ion-icon :icon="closeCircleOutline" color="danger" v-if="failedJobs.includes(job.jobId)" />
+        <ion-reorder />
+      </ion-item>
+    </ion-reorder-group>
+
+    <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+      <ion-fab-button @click="save()" :disabled="this.failedJobs.length">
```
Comment (adityasharma7) [inline src/components/JobReorderModal.vue:26]:
```suggestion
      <ion-fab-button @click="save()" :disabled="failedJobs.length">
```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**